shared/elements/gaia-site-icon/script.js

```diff
@@ -279,7 +280,7 @@ window.GaiaAppIcon = (function(exports) {
         this._image = null;
         this._hasIcon = true;
 
-        if (!this._hasPredefinedIcon && !this._hasUserSetIcon) {
+        if (!this._hasPredefinedIcon) {
```

Comment:
Alberto wanted this change for system, but we need to handle it in homescreen - icon-loaded will always be fired now, even if the icon was set by the icon property. This means that the homescreen will unnecessarily re-save every icon on startup, which would be best avoided.